### PR TITLE
chore: use luarocks-tag-release workflow

### PR DESCRIPTION
**Pull request by mrcjkb**

### CODE REVIEW

File: nvim-lspconfig-scm-1.rockspec [hunk 1]
```diff
@@ -20,16 +20,9 @@ dependencies = {
 }
 
 source = {
-  url = 'https://github.com/neovim/nvim-lspconfig/archive/v' .. _MODREV .. '.zip',
-  dir = 'nvim-lspconfig-' .. _MODREV,
+  url = 'git://github.com/neovim/nvim-lspconfig',
```
Comment (teto) [inline nvim-lspconfig-scm-1.rockspec:23]: have you tested this pattern work with tagged release too ?

Comment (mrcjkb) [reply nvim-lspconfig-scm-1.rockspec:23]: The luarocks-tag-release action doesn't use the `scm-1.rockspec` so there's no need to handle sources for tagged releases in this file.

File: README.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ See also `:help lsp-config`.
 
 ## Install
 
+[![LuaRocks](https://img.shields.io/luarocks/v/teto/nvim-lspconfig?logo=lua&color=purple)](https://luarocks.org/modules/teto/nvim-lspconfig)
```
Comment (teto) [inline README.md:13]: this is now released by the "neovim" user can you update the urls ? https://luarocks.org/modules/neovim/nvim-lspconfig 
then I'll merge

**Comment by teto** [comment]
thanks again for the huge work you've put into this. I'll try to help maintain the momentum ;)

**Comment by mrcjkb** [comment]
> thanks again for the huge work you've put into this. I'll try to help maintain the momentum ;)

No problem :)
I'd love to see the plugin ecosystem improve and am glad to help where I can.

**Comment by teto** [comment]
seems like there is a bug in teh generated rockspec yet upload succeded https://github.com/neovim/nvim-lspconfig/actions/runs/4098805462/jobs/7068176967#step:4:329

**Comment by mrcjkb** [comment]
It looks like there's a bug in the github action. It should escape `'` in strings.

**Pull request merged**